src/config/loading/schema.rs

```diff
@@ -0,0 +1,80 @@
+use super::ConfigBuilder;
+
+const SECRET_AND_VARIABLES_REGEX: &str = "(:?\\$\\{.+\\})";
+
+fn into_errors<E: std::fmt::Debug>(err: E) -> Vec<String> {
+    vec![format!("{:?}", err)]
+}
+
+fn generate_schema() -> Result<serde_json::Value, Vec<String>> {
+    let schema =
+        vector_config::schema::generate_root_schema::<ConfigBuilder>().map_err(into_errors)?;
+
+    let mut schema = serde_json::to_value(&schema).map_err(into_errors)?;
+    // add format key to enforce check
+    if let Some(def) = schema
+        .as_object_mut()
+        .and_then(|r| r.get_mut("definitions"))
+        .and_then(|d| d.as_object_mut())
+        .and_then(|d| d.get_mut("vector_common::sensitive_string::SensitiveString"))
```

Comment:
Just to make this a little more stable, use `<SensitiveString as Configurable>::referenceable_name()` instead of the hard-coded string.

You should be able to call `expect` on the result (which is `Option<&'static str>`) because we know that we expect it to have a unique, referenceable name (i.e. it isn't just a basic type like a string or integer).